tests/flows/erc20_source_multihop.go

```diff
@@ -0,0 +1,164 @@
+package flows
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/ava-labs/subnet-evm/accounts/abi/bind"
+	erc20source "github.com/ava-labs/teleporter-token-bridge/abi-bindings/go/ERC20Source"
+	"github.com/ava-labs/teleporter-token-bridge/tests/utils"
+	"github.com/ava-labs/teleporter/tests/interfaces"
+	teleporterUtils "github.com/ava-labs/teleporter/tests/utils"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	. "github.com/onsi/gomega"
+)
+
+func ERC20SourceMultihop(network interfaces.Network) {
+	/**
+	 * Deploy a erc20 token source on the primary network
+	 * Deploys ERC20Destination to Subnet A and Subnet B
+	 * Bridges C-chain example erc20 tokens to Subnet A
+	 * Bridge tokens from Subnet A to Subnet B through multihop
+	 * Brige back tokens from Subnet B to Subnet A through multihop
+	 */
+
+	cChainInfo := network.GetPrimaryNetworkInfo()
+	subnetAInfo, subnetBInfo := teleporterUtils.GetTwoSubnets(network)
+	fundedAddress, fundedKey := network.GetFundedAccountInfo()
+
+	ctx := context.Background()
+
+	// Deploy an ExampleERC20 on subnet A as the source token to be bridged
+	sourceTokenAddress, sourceToken := teleporterUtils.DeployExampleERC20(
+		ctx,
+		fundedKey,
+		cChainInfo,
+	)
+
+	// Create an ERC20Source for bridging the source token
+	erc20SourceAddress, erc20Source := utils.DeployERC20Source(
+		ctx,
+		fundedKey,
+		cChainInfo,
+		fundedAddress,
+		sourceTokenAddress,
+	)
+
+	// Token representation on subnet B will have same name, symbol, and decimals
+	tokenName, err := sourceToken.Name(&bind.CallOpts{})
+	Expect(err).Should(BeNil())
+	tokenSymbol, err := sourceToken.Symbol(&bind.CallOpts{})
+	Expect(err).Should(BeNil())
+	tokenDecimals, err := sourceToken.Decimals(&bind.CallOpts{})
+	Expect(err).Should(BeNil())
+
+	// Deploy an ERC20Destination for Subnet A
+	erc20DestinationAddress_A, erc20Destination_A := utils.DeployERC20Destination(
```

Comment:
```suggestion
	erc20DestinationAddressA, erc20DestinationA := utils.DeployERC20Destination(
```

I think the underscore there is not standard golang variable naming convention. (Same for _B variables below)

Reply:
Generally I'd agree to follow the convention, but here since it's for tests, and the variable naming is so long and similar, I'd prefer better readability. Think it's easy to mix up the two with golang convention here.

Reply:
for example, just realized I made that mistake by specifying wrong chain's bridge address

Reply:
I don't have a strong opinion especially given it's test, but generally think if a name doesn't work will in a given convention, it makes more sense to re-evaluate the name than break the convention 🙂

Reply:
that's reasonable, will revisit after we have more e2e tests